### ISSUE TITLE
Poor results training YOLO v3 on PASCAL VOC data

### ISSUE THREAD

**Issue by ayooshkathuria** [pjreddie/darknet#1264]
I've been trying to train YOLO v3 on VOC dataset. I follow the exact train instructions given on the website to train the data. I'm training on a Titan Xp using a batch size of 64 and subdivision of 16. I let the network train fully, for 50200 iterations. 

But the results are really bad. I cannot breach even 20 on the mAP whereas YOLO v2 gave me 75.68 on the same metric. 

Any one with any suggestions?

**Comment by Hello526** [comment]
@ayooshkathuria Yes, it is true. I set batch=64, subdivision=16, trained 40,000 batches on VOC2007+2012, and only got 20~30% of mAP on VOC2007 test.

**Comment by 0801130205** [comment]
@ayooshkathuria @Hello526 I also get the problem. The loss is around 0.19 on the 45978th iteration,but the prediction is so bad.
![Screenshot from 2019-03-31 09-42-56](https://user-images.githubusercontent.com/29110881/55283607-9667d680-5399-11e9-9bb3-6d11c58c5835.png)

**Comment by 0801130205** [comment]
@ayooshkathuria @Hello526 I also get the problem. The loss is around 0.19 on the 45978th iteration,but the prediction is so bad.

loss:
![Screenshot from 2019-03-31 09-42-56](https://user-images.githubusercontent.com/29110881/55283621-de86f900-5399-11e9-9958-05ee049e8500.png)

prediction(./darknet detect cfg/yolov3-voc.cfg backup/yolov3-voc.backup scripts/VOCdevkit/VOC2007/JPEGImages/007040.jpg):
![1](https://user-images.githubusercontent.com/29110881/55283634-0fffc480-539a-11e9-8d10-90f190ddf382.png)
![2](https://user-images.githubusercontent.com/29110881/55283640-34f43780-539a-11e9-9e93-75fd020a6af2.png)
![3](https://user-images.githubusercontent.com/29110881/55283652-6c62e400-539a-11e9-8f8b-de7e0e2adf60.png)

**Comment by joaolcaas** [comment]
@0801130205 
> @ayooshkathuria @Hello526 I also get the problem. The loss is around 0.19 on the 45978th iteration,but the prediction is so bad.

This is not an error on your model. If you check coco.names and voc.names, bird and person it is about 15th position on file. So, my gess is that you are using coco.names otherwise voc.names. Check a place that you can change it.

**Comment by 0801130205** [comment]
@joaolcaas Thank you for your response. I have solved the problem that is truely caused by what you said.

**Comment by irishjars** [comment]
Hi @0801130205  where did you make this change?

**Comment by joaolcaas** [comment]
I dont record exactly. But i searched for "coco.names" in code and i made the change

**Comment by reactivetype** [comment]
@joaolcaas  @081130205 can you report on the new mAP you get on VOC2007?

**Comment by joaolcaas** [comment]
Sry @reactivetype, i dot not have this data anymore.

**Comment by jesliyauvionics** [comment]
> I dont record exactly. But i searched for "coco.names" in code and i made the change

Which code did you change? can you please explain?
